ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/configuration/LedgerIdRequirement.scala

```diff
@@ -4,11 +4,16 @@
 package com.daml.ledger.client.configuration
 
 /**
-  * @param ledgerId The ID of the target ledger.
-  * @param enabled If true, the client will only communicate with ledgers that have the expected LedgerId.
-  *                Note that this setting only affects the binding process, when the ledger ID on the server is checked.
+  * @param ledgerId The ID of the target ledger. If defined, the client will only communicate with ledgers that have the
+  *                 expected LedgerId. Note that this setting only affects the binding process, when the ledger ID
+  *                 on the server is checked.
   */
-final case class LedgerIdRequirement(ledgerId: String, enabled: Boolean) {
+final case class LedgerIdRequirement(ledgerId: Option[String]) {
```

Comment:
Reading this now, I think I'd prefer to keep this constructor private and add a couple of static factory methods:

```scala
object LedgerIdRequirement {
  val none = LedgerIdRequirement(None)

  def matching(ledgerId: String) = LedgerIdRequirement(Some(ledgerId))

  @deprecated("Use static factory methods", "1.3.0")
  def apply(ledgerId: String, enabled: Boolean): LedgerIdRequirement =
    if (enabled) matching(ledgerId) else none
}
```

Reply:
That constructor cannot be private or it will break the ASX build.

Reply:
I thought existing code will be using the new `def apply`, which would stay public?

Reply:
I've added the factory methods but also kept the option based constructor as that is often easier when constructing from config.